src/main/scala/edu/cmu/cs/obsidian/parser/Parser.scala

```diff
@@ -463,8 +463,10 @@ object Parser extends Parsers {
                     case None =>
                         Field(isConst = false, typ, name._1, availableInSet).setLoc(typ)
             }
+            case _ ~ _ ~ _ ~ _ ~ Some(x) ~ _ => throw new RuntimeException("TODO") //failure("available in fields only permitted at the top level of contracts")
```

Comment:
This is a weird comment here — the text in the failure call is not a TODO, it's just not supported. Is that what's going on here? If so, replace TODO with that comment.

Reply:
addressed in https://github.com/mcoblenz/Obsidian/pull/301/commits/ec413aeb0ca0e52c33d8bb8cc506a5ece96a90bc !